cluster-logging.md

```diff
@@ -0,0 +1,110 @@
+
+**Note:** Logs are forwarded using OTLP/HTTP as defined by the OpenTelemetry Observability Framework.  It uses Protobuf payloads encoded in JSON format.
+
+
+## Semantic Conventions
+
+The log collector provided by this solution collects the following log streams:
+
+* Container logs
+* Cluster node journal logs
+* Cluster node auditd logs 
+* Kubernetes and OpenShift API server logs
+* OpenShift Virtual Network (OVN) logs
+
+These streams are forwarded using the sementatic conventions defined by [OpenTelemetry semantic attributes](https://github.com/open-telemetry/semantic-conventions/tree/main/docs). The semantic conventions in OpenTelemetry define a *Resource* as an immutable representation of the entity producing telemetry as *Attributes*. For example, a process producing telemetry that is running in a container has a container_name, a cluster_id, a pod_name, a namespace, and possibly a deployment or app_name. All of these *Attributes* are included in the *Resource* object.  This grouping and reducing of common attributes is a powerful tool when sending logs as telemetry data.
+
+
+The following sections define the attributes that are forwarded.
+
+
+### Common Attributes
+
+All log streams include a minimal set of attributes.
+
+Resource Attributes:
+```
+cluster.id
+node.name
+openshift.log.source
+openshift.log.type
+```
+
+Log Attributes:
+```
+body
+timeUnixNano
+severityNumber
+observedTimeUnixNano
+```
+
+### Kubernetes Container Logs
+
+Resource Attributes:
+```
+k8s.cluster.uid
+k8s.container.name
+k8s.namespace.name
+k8s.pod.name
+```
+
+Log Attributes:
+```
+k8s.container.id
+k8s.node.name
+k8s.pod.uid
+```
+
+The following attributes, along with their OTEL equivalents, are added to support minimal backwards compatibility with the [ViaQ](https://github.com/openshift/cluster-logging-operator/blob/release-6.0/docs/reference/datamodels/viaq/v1.adoc) data model. These attributes should be considered deprecated
+and will be removed in a future update.
+
+```
+kubernetes.container_name -> k8s.container.name
+kubernetes.namespace_name -> k8s.namespace.name
+kubernetes.pod_name       -> k8s.pod.name
+log_source                -> openshift.log.source
+log_type                  -> openshift.log.type
+```
+
+### Kubernetes and OpenShift API, OVN Logs
+
+Log Attributes:
+```
+http.response.status.code
+http.request.method
+url.full
+```
+
+### Cluster Node Journal Logs
+
+Log Attributes:
+```
+system.cgroup
+system.cmdline
+syslog.facility
+syslog.identifier
+system.invocation.id
+syslog.procid
+system.slice
+system.unit
```

Comment:
I am missing the following that I found in our log examples:
```
service.name
host.name
process.command
```

Reply:
Reviewing the actual vector transform it appears the attributes you identify are part of the payload and most of the ones I listed are not.  Updating to reflect the transform.  We can determine afterwards if there needs to be changes

Reply:
Correction.  We have "explicit" transformations and we also have ones we dynamically create based upon the existing fields.  I will further review

Reply:
I need to investigate because my original statement looks to be accurate but doesn't reflect the samples we have committed.

Reply:
Updated text based upon what I found in the actual transform